lib/models/singleton-addon-cache/cache-entry.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+/**
+ * Class for the entries in the SingletonAddonCache. The cache itself is keyed
+ * on the addon's PackageInfo.root.
+ *
+ * @class CacheEntry
+ */
+class CacheEntry {
```

Comment:
Hmm, do we really need this class? Seems like we can handle this shaping concern (built or not)  at callsites.

Reply:
Good idea - removing in next version.

Reply:
Actually, we do need some sort of 'wrapper' around the instance value (though we don't need the 'isBuilt' method) because we have to have an object for a proxy to point at, potentially before an instance actually exists. In the case where an addon instance is referenced by both a lazy engine and the project, the project's one takes precedence. But if we encounter the lazy-engine dependency first, we need the engine to point to a proxy and then the project to fill in the proxy instance later when we get to that dependency processing.